clang/docs/LanguageExtensions.rst

```diff
@@ -434,6 +434,36 @@ __datasizeof
 ``__datasizeof`` behaves like ``sizeof``, except that it returns the size of the
 type ignoring tail padding.
 
+.. _builtin_structured_binding_size-doc:
+
+__builtin_structured_binding_size (C++)
+---------------------------------------
+``__builtin_structured_binding_size`` returns the *structured binding size*
+([dcl.struct.bind]) of the type ``T`` (or unevaluate expression ``arg``)
+passed as argument.
+
+This is equivalent to the size of the pack ``p`` in ``auto&& [...p] = arg;``.
+If the argument is not destructurable (ie not an array, vector, complex,
```

Comment:
Is `vector` really destructurable?

Reply:
I guess we should mention built-in array here.

Reply:
I guess I should be more specific here.
I meant built in arrays and simd vectors, not the STL types
https://clang.llvm.org/docs/LanguageExtensions.html#vectors-and-extended-vectors